src/resources/2017/Spacegate.ts

```diff
@@ -0,0 +1,155 @@
+import { cliExecute, Item, visitUrl } from "kolmafia";
+import { get } from "../../property";
+import { $item, $items } from "../../template-string";
+
+export function have(): boolean {
+  return get("spacegateAlways") || get("_spacegateToday");
+}
+
+export function checkStatus(): void {
+  visitUrl("place.php?whichplace=spacegate&action=sg_Terminal");
+}
+
+export function dialled(): boolean {
+  checkStatus();
+  return get("_spacegateCoordinates") !== "";
+}
+
+export function hazards() {
+  return get("_spacegateHazards");
+}
+
+export function planetName() {
+  return get("_spacegatePlanetName");
+}
+
+export function planetCoords() {
+  return get("_spacegateCoordinates");
+}
+
+export function plantLife() {
+  return get("_spacegatePlantLife");
+}
+
+export function animalLife() {
+  return get("_spacegateAnimalLife");
+}
+
+export function intelligentLife() {
+  return get("_spacegateIntelligentLife");
+}
+
+export function ruins() {
+  return get("_spacegateRuins");
+}
+
+export function spants() {
+  return get("_spacegateSpant");
+}
+
+export function murderBots() {
+  return get("_spacegateMurderbot");
+}
+
+export function hazardEquipment(hazards: string): Item | Item[] {
+  const equipment: Item[] = $items`none`;
+  if (hazards.includes("toxic atmosphere"))
+    equipment.push($item`filter helmet`);
+  if (hazards.includes("high gravity"))
+    equipment.push($item`exo-servo leg braces`);
+  if (hazards.includes("irradiated")) equipment.push($item`rad cloak`);
+  if (hazards.includes("magnetic storms"))
+    equipment.push($item`gate transceiver`);
+  if (hazards.includes("high winds"))
+    equipment.push($item`high-friction boots`);
+  if (equipment.length === 2) {
+    return equipment[1];
+  } else {
+    return equipment;
+  }
+}
+
+export function getHazardEquipment(): void {
+  const equipment = hazardEquipment(hazards());
+
+  if (Array.isArray(equipment)) {
+    equipment.forEach((equip) => {
+      const num = () => {
+        switch (equip) {
+          case $item`filter helmet`:
+            return 1;
+          case $item`exo-servo leg braces`:
+            return 2;
+          case $item`rad cloak`:
+            return 3;
+          case $item`gate transceiver`:
+            return 4;
+          case $item`high-friction boots`:
+            return 5;
+        }
+        visitUrl("place.php?whichplace=spacegate&action=sg_requisition");
+        visitUrl(`choice.php?whichchoice=1233&option=${num}`);
+      };
+    });
+  } else {
+    const num = () => {
+      switch (equipment) {
+        case $item`filter helmet`:
+          return 1;
+        case $item`exo-servo leg braces`:
+          return 2;
+        case $item`rad cloak`:
+          return 3;
+        case $item`gate transceiver`:
+          return 4;
+        case $item`high-friction boots`:
+          return 5;
+      }
```

Comment:
This is repeated twice, which means it should precede the actual condition, and at the same time, given that you never invoke the function `num`, I'm pretty sure the visitUrls it's in aren't going to work.

Instead, consider defining a Map, or putting them all in an array and adding something to indexOf, or doing toInt to the equipment and subtracting something from it.